apps/intellij/src/main/kotlin/dev/nx/console/ide/project_json_inspection/AnalyzeNxConfigurationFilesAction.kt

```diff
@@ -25,15 +27,20 @@ class AnalyzeNxConfigurationFilesNotificationAction :
     }
 }
 
+@Suppress("UnstableApiUsage")
 fun checkForCodeSmells(project: Project) {
-    val codeSmellDetector = CodeSmellDetector.getInstance(project)
-    val codeSmells = codeSmellDetector.findCodeSmells(findNxConfigurationFiles(project))
-    if (codeSmells.size == 0) {
-        Notifier.notifyAnything(
-            project,
-            "We couldn't find any problems in your Nx configuration files. Is the \$schema property specified?"
-        )
-    } else {
-        codeSmellDetector.showCodeSmellErrors(codeSmells)
+    runWithModalProgressBlocking(project, "Check for code smells") {
```

Comment:
@MaxKless - you were asking about how one should go to coroutines context from the action. I think this one is a good starting point if the action is performing some stuff in the background and user should see the progress.